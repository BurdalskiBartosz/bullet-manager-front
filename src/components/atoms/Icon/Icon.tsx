import { FC } from 'react';
import icons from 'assets/svgs';

export type tIconNames =
	| 'password'
	| 'show_password'
	| 'magnifier'
	| 'arrow_right'
	| 'add'
	| 'goal'
	| 'note'
	| 'project'
	| 'schedule'
	| 'list'
	| 'account'
	| 'settings'
	| 'logout'
	| 'task'
	| 'close';

export type tIcon = {
	iconName: tIconNames;
	fill?: string;
	width?: string;
	height?: string;
	className?: string;
};

const Icon: FC<tIcon> = ({
	iconName,
	fill,
	width = '20px',
	height = '20px',
	className
}) => {
	const IconComponent = icons[iconName];

	if (!IconComponent) return null;

	return (
		<IconComponent
			className={className}
			fill={fill}
			width={width}
			height={height}
		/>
	);
};

export default Icon;
