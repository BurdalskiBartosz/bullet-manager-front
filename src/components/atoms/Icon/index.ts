export { default } from './Icon';
export type { tIconNames } from './Icon';
