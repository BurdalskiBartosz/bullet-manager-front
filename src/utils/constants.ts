export const baseURL: string = 'http://localhost:3001/api';

export const appRootRoute: string = 'app';
