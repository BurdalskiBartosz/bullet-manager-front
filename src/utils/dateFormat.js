import { format } from 'date-fns';

export const dateFormat = (value) => format(value, 'MM/dd/yyyy');
