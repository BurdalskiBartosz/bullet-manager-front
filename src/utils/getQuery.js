export const getQuery = (object) => {
	if (!object) return '';
	const queryStrings = Object.entries(object).map(([key, value]) => {
		const stringKey = encodeURIComponent(key);
		if (typeof value === 'object')
			return `${stringKey}=${JSON.stringify(value)}`;
		else return `${stringKey}=${encodeURIComponent(value)}`;
	});

	if (queryStrings.length === 0) return '';
	return '?' + queryStrings.join('&');
};
