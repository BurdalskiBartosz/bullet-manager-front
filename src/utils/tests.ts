import { render } from '@testing-library/react';
import Providers from '../providers';

const customRender = (ui: any) => render(ui, { wrapper: Providers });

export * from '@testing-library/react';

export { customRender as render };
