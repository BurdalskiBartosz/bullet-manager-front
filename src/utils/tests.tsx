import { render, RenderOptions } from '@testing-library/react';
import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { AppStore, RootState, setupStore } from 'store/store';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { BrowserRouter } from 'react-router-dom';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}
const reduxStore = setupStore();
export const renderWithProviders = (
	ui: React.ReactElement,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = reduxStore,
		...renderOptions
	}: ExtendedRenderOptions = {}
) => {
	const Wrapper = ({ children }: PropsWithChildren<{}>) => {
		return (
			<BrowserRouter>
				<Provider store={store}>
					<ThemeProvider theme={theme}>{children}</ThemeProvider>
				</Provider>
			</BrowserRouter>
		);
	};

	// Return an object with the store and all of RTL's query functions
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders as render };
export * from '@testing-library/react';
