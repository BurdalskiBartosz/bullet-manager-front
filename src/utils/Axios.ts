/* eslint-disable no-param-reassign */
import axios, { AxiosRequestConfig, AxiosError, AxiosResponse } from 'axios';
import { baseURL } from './constants';

const options: AxiosRequestConfig = {
	baseURL,
	timeout: 10000,
	headers: { 'Content-Type': 'application/json' },
	withCredentials: true
};

export const fetchProvider = axios.create(options);
const { interceptors } = fetchProvider;

const getRequestConfig = (config: AxiosRequestConfig) => {
	const token: string | null = window.localStorage.getItem('token');

	if (token) config.headers!.Authorization = `Bearer ${token}`;

	return config;
};

const getRequestError = (error: AxiosError) => Promise.reject(error);

const getResponse = (response: AxiosResponse) => response;

const getResponseError = (error: AxiosError) => {
	const errorResponse =
		error.response?.status === 400 ? error : error.response;
	return Promise.reject(errorResponse);
};

interceptors.request.use(getRequestConfig, getRequestError);
interceptors.response.use(getResponse, getResponseError);

export const instanceForMocks = axios.create(options);
