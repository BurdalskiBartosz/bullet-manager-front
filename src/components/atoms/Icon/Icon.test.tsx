import { render, screen } from 'utils/tests';
import Icon from './Icon';

describe('Icon', () => {
	it('should render correctly', () => {
		render(<Icon iconName="close" />);
		const iconElement = screen.getByTestId('icon-component');
		expect(iconElement).toMatchSnapshot();
	});
});
