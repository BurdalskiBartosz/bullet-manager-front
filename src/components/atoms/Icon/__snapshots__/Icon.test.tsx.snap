// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<svg
  data-testid="icon-component"
  height="20px"
  width="20px"
>
  close.svg
</svg>
`;
